### PR TITLE
chore: Remove reconciler prow jobs

### PR DESCRIPTION
**Pull request by c-pius**
**Description**

Changes proposed in this pull request:

- removes prow jobs configs and related scripts for reconciler

**Related issue(s)**

- resolves https://github.com/kyma-project/test-infra/issues/9873

### CODE REVIEW

File: pkg/tools/pjtester/test_artifacts/test-prow-config.yaml [hunk 1]
```diff
@@ -240,14 +240,7 @@ branch-protection:
...
                 strict: true
         reconciler:
-          protect: false
-          branches:
-            main:
-              protect: true
-            develop:
-              protect: true
-            load-test-mock:
-              protect: true
```
Comment (jeremyharisch) [inline pkg/tools/pjtester/test_artifacts/test-prow-config.yaml:250]: I would only remove the branch protection if the reconciler repo is archived.

File: prow/config.yaml [hunk 2]
```diff
@@ -398,16 +398,7 @@ branch-protection:
...
-          protect: false
-          branches:
-            main:
-              protect: true
-            develop:
-              protect: true
-            load-test-mock:
-              protect: true
-            istio-op:
-              protect: true
```
Comment (jeremyharisch) [inline prow/config.yaml:410]: Same here: I would only remove the branch protection if the reconciler repo is archived.

File: prow/config.yaml [hunk 3]
```diff
@@ -447,7 +438,6 @@ tide:
     kyma-project/modulectl: squash
     kyma-project/cloud-manager: squash
     kyma-incubator/compass: squash
-    kyma-incubator/reconciler: squash
```
Comment (jeremyharisch) [inline prow/config.yaml:450]: Squah should still be active if some PRs are getting created

File: pkg/tools/pjtester/test_artifacts/test-prow-config.yaml [hunk 4]
```diff
@@ -268,7 +261,6 @@ tide:
     kyma-project/kyma-watcher: squash
     kyma-project/manifest-operator: squash
     kyma-incubator/compass: squash
-    kyma-incubator/reconciler: squash
```
Comment (jeremyharisch) [inline pkg/tools/pjtester/test_artifacts/test-prow-config.yaml:271]: Squash should still be active if some PRs are getting created

**Review by jeremyharisch — COMMENTED** [review]
Removing all the jobs looks good. But since the reconciler repository is not archived until now, I would vote for not deleting the branch protections and squash merge rules

**Comment by c-pius** [comment]
Didn't use the fork by accident. Closing this in favor of a PR from the forked repository: https://github.com/kyma-project/test-infra/pull/11051

**Pull request closed without merging**